gptme/server/api_v2_sessions.py

```diff
@@ -295,6 +346,14 @@ def step(
         _append_and_notify(manager, session, msg)
         logger.debug("Persisted assistant message")
 
+        # Trigger MESSAGE_POST_PROCESS hook
+        if post_msgs := trigger_hook(
+            HookType.MESSAGE_POST_PROCESS,
+            manager=manager,
+        ):
+            for msg in post_msgs:
+                _append_and_notify(manager, session, msg)
```

Comment:
This seems to reassign `msg` later used in `"message": msg2dict(msg, manager.workspace)`

Explains the failing test?